ml-agents/mlagents/trainers/meta_curriculum.py

```diff
@@ -115,16 +116,24 @@ def increment_lessons(self, measure_vals, reward_buff_sizes=None):
                 )
         return ret
 
-    def set_all_curricula_to_lesson_num(self, lesson_num):
-        """Sets all the curricula in this meta curriculum to a specified
-        lesson number.
-
-        Args:
-            lesson_num (int): The lesson number which all the curricula will
-                be set to.
+    def try_restore_all_curriculum(self):
         """
-        for _, curriculum in self.brains_to_curricula.items():
-            curriculum.lesson_num = lesson_num
+        Tries to restore all the curriculums to what is saved in training_status.json
+        """
+
+        for brain_name, curriculum in self.brains_to_curricula.items():
+            # Create a temporary StatsReporter with the right brain name
+            _statsreporter = StatsReporter(brain_name)
```

Comment:
It's unclear what's going on here; it feels really hacky (and probably brittle).

Reply:
It's less of a concern now since we're no longer using StatsReporters, but we're still using the `brain_name` to refer to meta curriculums.